homeassistant/components/shelly/utils.py

```diff
@@ -462,3 +463,20 @@ def is_rpc_wifi_stations_disabled(
         return False
 
     return True
+
+
+async def parse_host(input_host: str) -> tuple[str, int]:
+    """Parse host to verify port."""
+
+    if ":" in input_host:
+        splited: list = input_host.split(":")
+        try:
+            host = splited[0]
+            port = int(splited[1])
+        except ValueError as ex:
+            raise HomeAssistantError("Syntax error") from ex
+    else:
+        host = input_host
+        port = 80
+
+    return (host, port)
```

Comment:
We shouldn't be using a custom parser for config flow inputs. The config flow uses `voluptuous` to validate user fields and fields should be visible to the user which will also make it easier to document them.

Example (all other integrations with HOST/PORT are the same):
https://github.com/home-assistant/core/blob/34d316e7b5b4465aaddcefa2d58dae02a9bb9afe/homeassistant/components/atag/config_flow.py#L14-L17
https://www.home-assistant.io/integrations/atag/#configuration-variables

Reply:
Was discussed with @bdraco and due to the fact that that less users will use this config was suggested to not expose the port field to the flow